src/App.tsx

```diff
@@ -1,16 +1,47 @@
-import React from 'react';
+import React, { useCallback, useMemo, useState } from 'react';
 import './App.scss';
 import { peopleFromServer } from './data/people';
 
+import debounce from 'lodash.debounce';
+import cn from 'classnames';
+import { Person } from './types/Person';
+
 export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+  const [isFocused, setIsFocused] = useState(false);
+  const [query, setQuery] = useState('');
+  const [selectedPerson, setSelectedPerson] = useState<Person | null>(null);
+  const [appliedQuery, setAppliedQuery] = useState('');
+
+  const applyQuery = useCallback(debounce(setAppliedQuery, 1000), []);
+
+  const handleQuery = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setSelectedPerson(null);
+    setQuery(event.target.value);
+    applyQuery(event.target.value);
```

Comment:
You are not checking if the user entered spaces only before calling the filter. Consider adding a check for this case to avoid unnecessary filtering.